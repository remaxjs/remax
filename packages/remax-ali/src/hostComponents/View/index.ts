import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface ViewProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  disableScroll?: boolean;
  hoverClassName?: string;
  hoverStartTime?: number;
  hoverStayTime?: number;
  hidden?: boolean;
  animation?: any;
  hoverStopPropagation?: boolean;
  slot?: string;
  onTap?: (e: any) => void;
  catchTap?: (e: any) => void;
  onClick?: (e: any) => void;
  catchClick?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
  onLongTap?: (e: any) => void;
  onTransitionEnd?: (e: any) => void;
  onAnimationIteration?: (e: any) => void;
  onAnimationStart?: (e: any) => void;
  onAnimationEnd?: (e: any) => void;
  onAppear?: (e: any) => void;
  onDisappear?: (e: any) => void;
  onFirstAppear?: (e: any) => void;
}

export const View = createHostComponent<ViewProps>('view') as React.ComponentType<ViewProps>;
