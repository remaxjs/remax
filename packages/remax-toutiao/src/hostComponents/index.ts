export * from './View';
export * from './Input';
export * from './Textarea';
export * from './Video';
export * from './Swiper';
export * from './ScrollView';
export * from './SwiperItem';
export * from './Icon';
export * from './Text';
export * from './RichText';
export * from './Progress';
export * from './Button';
export * from './CheckboxGroup';
export * from './Checkbox';
export * from './Form';
export * from './Label';
export * from './Picker';
export * from './PickerView';
export * from './PickerViewColumn';
export * from './RadioGroup';
export * from './Radio';
export * from './Slider';
export * from './Switch';
export * from './Navigator';
export * from './Image';
export * from './Canvas';
export * from './WebView';
export * from './MovableArea';
export * from './MovableView';
