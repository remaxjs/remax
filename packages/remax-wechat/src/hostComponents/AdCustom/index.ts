import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AdCustomProps extends BaseProps {
  /** 广告单元id，可在小程序管理后台的流量主模块新建 2.10.4; */
  unitId: string;
  /** 广告自动刷新的间隔时间，单位为秒，参数值必须大于等于30（该参数不传入时 Banner 广告不会自动刷新） 2.10.4 */
  AdIntervals?: number;
  /** 广告加载成功的回调 2.10.4 */
  onLoad?: (event: GenericEvent) => any;
  /** 广告加载失败的回调，event.detail = {errCode: 1002} 2.10.4 */
  onError?: (event: GenericEvent) => any;
}
/**
 * 原生模版广告
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/ad-custom.html
 */
export const AdCustom: React.ComponentType<AdCustomProps> = createHostComponent<AdCustomProps>('ad-custom');
