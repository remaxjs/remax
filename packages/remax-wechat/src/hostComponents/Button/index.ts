import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface ButtonProps extends BaseProps {
  /**
   * 1.0.0
   * default	默认大小
   * mini	小尺寸
   */
  size?: 'default' | 'mini';
  /**
   * 1.0.0
   * primary	绿色
   * default	白色
   * warn	红色
   */
  type?: 'primary' | 'default' | 'warn';

  /**
   * 1.0.0
   * 按钮是否镂空，背景色透明
   */
  plain?: boolean;
  /**
   * 1.0.0
   * 是否禁用
   */
  disabled?: boolean;
  /**
   * 1.0.0
   * 名称前是否带 loading 图标
   */
  loading?: boolean;
  /**
   * 1.0.0
   * submit	提交表单
   * reset	重置表单
   */
  formType?: 'submit' | 'reset';
  /**
   * - 微信小程序 & QQ 小程序
   * contact	打开客服会话，如果用户在会话中点击消息卡片后返回小程序，可以从 bindcontact 回调中获得具体信息	1.1.0
   * share	触发用户转发，使用前建议先阅读使用指引	1.2.0
   * getPhoneNumber	获取用户手机号，可以从 bindgetphonenumber 回调中获取到用户信息	1.2.0
   * getUserInfo	获取用户信息，可以从 bindgetuserinfo 回调中获取到用户信息	1.3.0
   * launchApp	打开 App，可以通过 app-parameter 属性设定向 App 传的参数	1.9.5
   * openSetting	打开授权设置页	2.0.7
   * feedback	打开“意见反馈”页面，用户可提交反馈内容并上传日志，开发者可以登录小程序管理后台后进入左侧菜单“客服反馈”页面获取到反馈内容	2.1.0
   *
   * - QQ 小程序
   * openGroupProfile	呼起群资料卡页面，可以通过 group-id 属性设定需要打开的群资料卡的群号，同时 app.json 中必须配置 groupIdList（数量不超过 10 个），表明可以打开群资料卡的群号	1.4.7
   * addFriend	添加好友，对方需要通过该小程序进行授权，允许被加好友后才能调用成功	1.0.0
   * addColorSign	添加彩签，点击后添加状态有用户提示，无回调	1.10.0
   * openPublicProfile	打开公众号资料卡，可以通过 public-id 属性设定需要打开的公众号资料卡的号码，同时 app.json 中必须配置 publicIdList（目前只能配置 1 个），表明可以打开的公众号资料卡的号码	1.12.0
   * addGroupApp	添加群应用（只有管理员或群主有权操作，建议先调用 qq.getGroupInfo 获取当前用户是否为管理员，如果是管理员才显示该按钮），添加后给 button 绑定 bindaddgroupapp 事件接收回调数据	1.16.0
   * shareMessageToFriend	在自定义开放数据域组件中,向指定好友发起分享	1.17.0
   * addToFavorites	收藏当前页面，点击按钮后会触发 Page.onAddToFavorites 方法	1.19.0
   */
  openType?:
    | 'contact'
    | 'share'
    | 'getPhoneNumber'
    | 'getUserInfo'
    | 'launchApp'
    | 'openSetting'
    | 'feedback'
    | 'openGroupProfile'
    | 'addFriend'
    | 'addColorSign'
    | 'openPublicProfile'
    | 'addGroupApp'
    | 'shareMessageToFriend'
    | 'addToFavorites';
  /**
   * 1.0.0
   * 指定按钮按下去的样式类。当 hover-class="none" 时，没有点击态效果
   */
  hoverClassName?: string;
  /**
   * 1.5.0
   * 指定是否阻止本节点的祖先节点出现点击态
   */
  hoverStopPropagation?: boolean;
  /**
   * 1.0.0
   * 按住后多久出现点击态，单位毫秒
   */
  hoverStartTime?: number;
  /**
   * 1.0.0
   * 手指松开后点击态保留时间，单位毫秒
   */
  hoverStayTime?: number;
  /**
   * 1.3.0
   * en	英文
   * zh_CN	简体中文
   * zh_TW	繁体中文
   */
  lang?: 'en' | 'zh_CN' | 'zh_TW';
  /**
   * 1.4.0
   * 会话来源，open-type="contact"时有效
   */
  sessionFrom?: string;
  /**
   * 1.5.0
   * 会话内消息卡片标题，open-type="contact"时有效
   */
  sendMessageTitle?: string;
  /**
   * 1.5.0
   * 会话内消息卡片点击跳转小程序路径，open-type="contact"时有效
   */
  sendMessagePath?: string;
  /**
   * 1.5.0
   * 会话内消息卡片图片，open-type="contact"时有效
   */
  sendMessageImg?: string;
  /**
   * 1.9.5
   * 打开 APP 时，向 APP 传递的参数，open-type=launchApp时有效
   */
  appParameter?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用的包名（Android）
   */
  appPackageName?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用的 Bundle ID（iOS）
   */
  appBundleID?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用在 QQ 互联中的 AppID
   */
  appConnectID?: string;
  /**
   * QQ 小程序 1.0.0
   * 添加好友时，对方的 openid
   */
  openId?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开群资料卡时，传递的群号
   */
  groupId?: string;
  /**
   * QQ 小程序 1.12.0
   * 打开公众号资料卡时，传递的号码
   */
  publicId?: string;
  /**
   * QQ 小程序 1.4.4
   * 分享类型集合，标志位形式
   */
  shareType?: number;
  /**
   * QQ 小程序 1.15.0
   * 分享类型集合，数组形式
   */
  shareMode?: Array<'qq' | 'qzone' | 'quickToDialog' | 'wechatFriends' | 'wechatMoment' | 'recentContacts'>;
  /**
   * QQ 小程序 1.17.0
   * 发送对象的 FriendInfo
   */
  shareMessageFriendInfo?: any;
  /**
   * QQ 小程序 1.17.0
   * 转发标题，不传则默认使用当前小程序的昵称。
   */
  shareMessageTitle?: string;
  /**
   * QQ 小程序 1.17.0
   * 转发显示图片的链接，可以是网络图片路径（仅 QQ CDN 域名路径）或本地图片文件路径或相对代码包根目录的图片文件路径。显示图片长宽比是 5:4
   */
  shareMessageImg?: string;
  /**
   * 1.5.0
   * 是否显示会话内消息卡片，设置此参数为 true，用户进入客服会话会在右下角显示"可能要发送的小程序"提示，
   * 用户点击后可以快速发送小程序消息，open-type="contact"时有效
   */
  showMessageCard?: boolean;
  /**
   * 1.3.0
   * 用户点击该按钮时，会返回获取到的用户信息，回调的detail数据与wx.getUserInfo返回的一致，
   * open-type="getUserInfo"时有效
   */
  onGetUserInfo?: (event: GenericEvent) => any;
  /**
   * 1.5.0
   * 客服消息回调，open-type="contact"时有效
   */
  onContact?: (event: GenericEvent) => any;
  /**
   * 1.2.0
   * 获取用户手机号回调，open-type=getPhoneNumber时有效
   */
  onGetPhoneNumber?: (event: GenericEvent) => any;
  /**
   * 2.21.2
   * 获取用户手机号回调，open-type=getPhoneNumber时有效
   */
  onChooseAvatar?: (event: GenericEvent) => any;
  /**
   * 1.9.5
   * 当使用开放能力时，发生错误的回调，open-type=launchApp时有效
   */
  onError?: (event: GenericEvent) => any;
  /**
   * 2.0.7
   * 在打开授权设置页后回调，open-type=openSetting时有效
   */
  onOpenSetting?: (event: GenericEvent) => any;
  /**
   * 2.4.4
   * 打开 APP 成功的回调，open-type=launchApp时有效
   */
  onLaunchApp?: (event: GenericEvent) => any;
  /**
   * QQ 小程序 1.0.0
   * 添加好友的回调
   */
  onAddFriend?: (event: GenericEvent) => any;
  /**
   * QQ 小程序 1.16.0
   * 添加群应用的回调
   */
  onAddGroupApp?: (event: GenericEvent) => any;
}

/**
 * https://developers.weixin.qq.com/miniprogram/dev/component/button.html
 * https://q.qq.com/wiki/develop/miniprogram/component/form/button.html
 */
export const Button: React.ComponentType<ButtonProps> = createHostComponent<ButtonProps>('button');

Button.defaultProps = {
  hoverClassName: 'button-hover',
  hoverStartTime: 20,
  hoverStayTime: 70,
};
