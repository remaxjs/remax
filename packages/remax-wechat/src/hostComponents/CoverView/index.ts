import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface CoverViewProps extends BaseProps {
  /**	设置顶部滚动偏移量，仅在设置了 overflow-y: scroll 成为滚动元素后生效	2.1.0  */
  scrollTop?: number | string;
  markerId?: string;
  slot?: string;
}

/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/cover-view.html
 */
export const CoverView: React.ComponentType<CoverViewProps> = createHostComponent<CoverViewProps>('cover-view');
