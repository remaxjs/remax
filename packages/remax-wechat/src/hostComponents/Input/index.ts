import * as React from 'react';
import { BaseProps, GenericEvent } from '../../types/component';
import { createHostComponent } from '@remax/runtime';

export interface InputProps extends BaseProps {
  /**
   * 1.0.0
   * (即将废弃，请直接使用 focus )自动聚焦，拉起键盘
   */
  autoFocus?: boolean;
  /**
   * 1.0.0
   * 获取焦点
   */
  focus?: boolean;
  /** 强制 input 处于同层状态，默认 focus 时 input 会切到非同层状态 (仅在 iOS 下生效) 2.10.4 */
  alwaysEmbed?: boolean;
  name?: string;
  /**
   * 1.0.0
   * 最大输入长度，设置为 -1 的时候不限制最大长度
   */
  maxlength?: number;
  /**
   * 1.0.0
   * 输入框的初始内容
   */
  value?: any;
  /**
   * 1.0.0
   * 是否是密码类型
   */
  password?: boolean;
  /**
   * 1.0.0
   * input 的类型
   *
   * text	文本输入键盘
   * number	数字输入键盘
   * idcard	身份证输入键盘
   * digit	带小数点的数字键盘
   * 2.18.0
   * safe-password 密码安全输入键盘
   * 2.21.2
   * nickname 	昵称输入键盘
   */
  type?: 'text' | 'number' | 'idcard' | 'digit' | 'safe-password' | 'nickname';
  /**
   * 1.1.0
   * 设置键盘右下角按钮的文字，仅在type='text'时生效
   *
   * send	右下角按钮为“发送”
   * search	右下角按钮为“搜索”
   * next	右下角按钮为“下一个”
   * go	右下角按钮为“前往”
   * done	右下角按钮为“完成”
   */
  confirmType?: 'send' | 'search' | 'next' | 'go' | 'done';
  /**
   * 1.1.0
   * 点击键盘右下角按钮时是否保持键盘不收起
   */
  confirmHold?: boolean;
  /**
   * 1.5.0
   * 指定focus时的光标位置
   */
  cursor?: number;
  /**
   * 1.9.0
   * 光标起始位置，自动聚集时有效，需与selection-end搭配使用
   */
  selectionStart?: number;
  /**
   * 1.9.0
   * 光标结束位置，自动聚集时有效，需与selection-start搭配使用
   */
  selectionEnd?: number;
  /**
   * 1.9.90
   * 键盘弹起时，是否自动上推页面
   */
  adjustPosition?: boolean;
  /**
   * 2.8.2
   * focus时，点击页面的时候不收起键盘
   */
  holdKeyboard?: boolean;
  /**
   * 1.0.0
   * 输入框为空时占位符
   */
  placeholder?: string;
  /**
   * 1.0.0
   * 指定 placeholder 的样式
   */
  placeholderStyle?: React.CSSProperties;
  /**
   * 1.0.0
   * 指定 placeholder 的样式类
   */
  placeholderClassName?: string;
  /**
   * 1.0.0
   * 是否禁用
   */
  disabled?: boolean;
  /**
   * 1.0.0
   * 指定光标与键盘的距离，取 input 距离底部的距离和 cursor-spacing 指定的距离的最小值作为光标与键盘的距离
   */
  cursorSpacing?: number;
  /**
   * 1.0.0
   * 键盘输入时触发，event.detail = {value, cursor, keyCode}，keyCode 为键值，
   * 2.1.0 起支持，处理函数可以直接 return 一个字符串，将替换输入框的内容。
   */
  onInput?: (event: GenericEvent) => void;
  /**
   * 1.0.0
   * 输入框聚焦时触发，event.detail = { value, height }，height 为键盘高度，在基础库 1.9.90 起支持
   */
  onFocus?: (event: GenericEvent) => void;
  /**
   * 1.0.0
   * 输入框失去焦点时触发，event.detail = {value: value}
   */
  onBlur?: (event: GenericEvent) => void;
  /**
   * 1.0.0
   * 点击完成按钮时触发，event.detail = {value: value}
   */
  onConfirm?: (event: GenericEvent) => any;
  /**
   * 2.7.0
   * 键盘高度发生变化的时候触发此事件，event.detail = {height: height, duration: duration}
   */
  onKeyboardHeightChange?: (event: GenericEvent) => any;
}

/**
 * https://developers.weixin.qq.com/miniprogram/dev/component/input.html
 */
export const Input: React.ComponentType<InputProps> = createHostComponent<InputProps>('input');

Input.defaultProps = {
  type: 'text',
  password: false,
  placeholderClassName: 'input-placeholder',
  disabled: false,
  cursorSpacing: 0,
  autoFocus: false,
  focus: false,
  alwaysEmbed: false,
  confirmType: 'done',
  confirmHold: false,
  maxlength: 140,
  selectionEnd: -1,
  selectionStart: -1,
  adjustPosition: true,
  holdKeyboard: false,
};
