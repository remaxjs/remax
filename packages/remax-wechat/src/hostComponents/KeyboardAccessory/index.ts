import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export type KeyboardAccessoryProps = BaseProps;

/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/keyboard-accessory.html
 */
export const KeyboardAccessory: React.ComponentType<KeyboardAccessoryProps> = createHostComponent<KeyboardAccessoryProps>(
  'keyboard-accessory'
);
