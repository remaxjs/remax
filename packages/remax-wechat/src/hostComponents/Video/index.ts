import * as React from 'react';
import { BaseProps, GenericEvent } from '../../types/component';
import { createHostComponent } from '@remax/runtime';

const hostComponentName = 'video';

export interface VideoProps extends BaseProps {
  /** 要播放视频的资源地址，支持云文件ID（2.3.0） 1.0.0 */
  src: string;
  /** 指定视频时长 1.1.0 */
  duration?: number;
  /** 是否显示默认播放控件（播放/暂停按钮、播放进度、时间） 1.0.0 */
  controls?: boolean;
  /** 弹幕列表 1.0.0 */
  danmuList?: any[];
  /** 是否显示弹幕按钮，只在初始化时有效，不能动态变更 1.0.0 */
  danmuBtn?: boolean;
  /** 是否展示弹幕，只在初始化时有效，不能动态变更 1.0.0 */
  enableDanmu?: boolean;
  /** 是否自动播放 1.0.0 */
  autoplay?: boolean;
  /** 是否循环播放 1.4.0 */
  loop?: boolean;
  /** 是否静音播放 1.4.0 */
  muted?: boolean;
  /** 指定视频初始播放位置 1.6.0 */
  initialTime?: number;
  /** 在非全屏模式下，是否开启亮度与音量调节手势（废弃，见 vslide-gesture） 1.6.0 */
  pageGesture?: boolean;
  /** 设置全屏时视频的方向，不指定则根据宽高比自动判断 1.7.0 */
  direction?: 0 | 90 | -90;
  /** 若不设置，宽度大于240时才会显示 1.9.0 */
  showProgress?: boolean;
  /** 是否显示全屏按钮 1.9.0 */
  showFullscreenBtn?: boolean;
  /** 是否显示视频底部控制栏的播放按钮 1.9.0 */
  showPlayBtn?: boolean;
  /** 是否显示视频中间的播放按钮 1.9.0 */
  showCenterPlayBtn?: boolean;
  /** 是否开启控制进度的手势 1.9.0 */
  enableProgressGesture?: boolean;
  /** 当视频大小与 video 容器大小不一致时，视频的表现形式 1.0.0 */
  objectFit?: 'contain' | 'fill' | 'cover';
  /** 视频封面的图片网络资源地址或云文件ID（2.3.0）。若 controls 属性值为 false 则设置 poster 无效 1.0.0 */
  poster?: string;
  /** 用于给搜索等场景作为视频封面展示，建议使用无播放 icon 的视频封面图，只支持网络地址 */
  posterForCrawler?: string;
  /** 显示投屏按钮。只安卓且同层渲染下生效，支持 DLNA 协议 1.10.2 */
  showCastingButton?: boolean;
  /** 设置小窗模式： push, pop，空字符串或通过数组形式设置多种模式（如： ["push", "pop"]） */
  pictureInPictureMode?: string | string[];
  /** 是否在小窗模式下显示播放进度	2.11.0 */
  pictureInPictureShowProgress?: boolean;
  /** 是否显示静音按钮 2.4.0 */
  showMuteBtn?: boolean;
  /** 视频的标题，全屏时在顶部展示 2.4.0 */
  title?: string;
  /** 播放按钮的位置 2.4.0 */
  playBtnPosition?: 'bottom' | 'center';
  /** 是否开启播放手势，即双击切换播放/暂停 2.4.0 */
  enablePlayGesture?: boolean;
  /** 是否开启手机横屏时自动全屏，当系统设置开启自动旋转时生效 2.11.0 */
  enableAutoRotation?: boolean;
  /** 是否显示锁屏按钮，仅在全屏时显示，锁屏后控制栏的操作 2.11.0 */
  showScreenLockButton?: false;
  /** 是否显示截屏按钮，仅在全屏时显示 2.13.0 */
  showSnapshotButton?: boolean;
  /** 是否展示后台音频播放按钮 2.14.3 */
  showBackgroundPlaybackButton?: boolean;
  /** 进入后台音频播放后的通知栏图标（Android 独有） */
  backgroundPoster?: string;
  /** 当跳转到其它小程序页面时，是否自动暂停本页面的视频 2.5.0 */
  autoPauseIfNavigate?: boolean;
  /** 当跳转到其它微信原生页面时，是否自动暂停本页面的视频 2.5.0 */
  autoPauseIfOpenNative?: boolean;
  /** 在非全屏模式下，是否开启亮度与音量调节手势（同 page-gesture） 2.6.2 */
  vslideGesture?: boolean;
  /** 在全屏模式下，是否开启亮度与音量调节手势 2.6.2 */
  vslideGestureInFullscreen?: boolean;
  /** 当开始/继续播放时触发play事件 1.0.0 */
  onPlay?: (event: GenericEvent) => any;
  /** 当暂停播放时触发 pause 事件 1.0.0 */
  onPause?: (event: GenericEvent) => any;
  /** 当播放到末尾时触发 ended 事件 1.0.0 */
  onEnded?: (event: GenericEvent) => any;
  /** 播放进度变化时触发，event.detail = {currentTime, duration} 。触发频率 250ms 一次 1.0.0 */
  onTimeUpdate?: (event: GenericEvent) => any;
  /** 视频进入和退出全屏时触发，event.detail = {fullScreen, direction}，direction 有效值为 vertical 或 horizontal 1.4.0 */
  onFullScreenChange?: (event: GenericEvent) => any;
  /** 视频出现缓冲时触发 1.7.0 */
  onWaiting?: (event: GenericEvent) => any;
  /** 视频播放出错时触发 1.7.0 */
  onError?: (event: GenericEvent) => any;
  /** 加载进度变化时触发，只支持一段加载。event.detail = {buffered}，百分比 2.4.0 */
  onProgress?: (event: GenericEvent) => any;
  /** 视频元数据加载完成时触发。event.detail = {width, height, duration} 2.7.0 */
  onLoadedMetaData?: (event: GenericEvent) => any;
  /** 切换 controls 显示隐藏时触发。event.detail = {show} 2.9.5 */
  onControlsToggle?: (event: GenericEvent) => any;
  /** 播放器进入小窗 2.11.0 */
  onEnterPictureInPicture?: (event: GenericEvent) => any;
  /** 播放器退出小窗 2.11.0 */
  onLeavePictureInPicture?: (event: GenericEvent) => any;
  /** seek 完成时触发 (position iOS 单位 s, Android 单位 ms) 2.12.0 */
  onSeekComplete?: (event: GenericEvent) => any;
}

const VideoRender: React.ForwardRefRenderFunction<any, VideoProps> = (props, ref) => {
  const { children, ...restProps } = props;
  const videoProps = {
    ...restProps,
    ref,
  };

  return React.createElement(hostComponentName, videoProps, children);
};

/**
 * video 默认宽度 300px、高度 225px，可通过 wxss 设置宽高
 * https://developers.weixin.qq.com/miniprogram/dev/component/video.html
 */
const RemaxVideo = React.forwardRef(VideoRender);

RemaxVideo.defaultProps = {
  controls: true,
  danmuBtn: false,
  enableDanmu: false,
  autoplay: false,
  loop: false,
  muted: false,
  initialTime: 0,
  pageGesture: false,
  showProgress: true,
  showFullscreenBtn: true,
  showPlayBtn: true,
  showCenterPlayBtn: true,
  enableProgressGesture: true,
  objectFit: 'contain',
  showMuteBtn: false,
  playBtnPosition: 'bottom',
  enablePlayGesture: false,
  autoPauseIfNavigate: true,
  autoPauseIfOpenNative: true,
  vslideGesture: false,
  vslideGestureInFullscreen: true,
  showCastingButton: false,
  pictureInPictureShowProgress: false,
  enableAutoRotation: false,
  showScreenLockButton: false,
  showSnapshotButton: false,
  showBackgroundPlaybackButton: false,
};

export const Video: React.ComponentType<VideoProps> = createHostComponent<VideoProps>(hostComponentName, RemaxVideo);
