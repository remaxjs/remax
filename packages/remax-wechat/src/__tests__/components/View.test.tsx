import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { View } from '../../hostComponents';

describe('View', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<View className="class">view</View>);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
