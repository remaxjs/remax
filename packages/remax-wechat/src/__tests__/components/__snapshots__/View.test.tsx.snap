// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View render correctly 1`] = `
<view
  className="class"
  hoverClassName="none"
  hoverStartTime={50}
  hoverStayTime={400}
  hoverStopPropagation={false}
>
  view
</view>
`;
